doc/source/dev/toolchain.rst

```diff
@@ -47,51 +47,64 @@ Python Versions
 ^^^^^^^^^^^^^^^
 
 SciPy is compatible with several versions of Python.  When dropping support for
-older Python versions, SciPy takes guidance from NEP 29 [1]_.  Python 2.7
-support was dropped starting from SciPy 1.3.
-
-================  =======================================================================
- Date             Pythons supported
-================  =======================================================================
- 2018              Py2.7, Py3.4+ (SciPy 1.2.x is the last release to support Python 2.7)
- 2019              Py3.5+ (but Py2.7-specific code not removed)
- 2020              Py3.6+ (removal of Py2.7-specific code permitted)
- 2021              Py3.7+
- 2022              Py3.8+
-================  =======================================================================
+older Python versions, SciPy takes guidance from [NEP29]_. Generally, support for
+the oldest Python version is dropped 42 months after the original release. Following
+the acceptance of PEP 602, this mostly happens in April, and gets picked up by the
+mid-year release of SciPy.
+
+.. dropdown:: Python version support over time
+
+    Python 2.7 support was dropped starting from SciPy 1.3.
+
+    ================  =======================================================================
+     Date             Pythons supported
+    ================  =======================================================================
+     2024              Py3.10+
+     2023              Py3.9+
+     2022              Py3.8+
+     2021              Py3.7+
+     2020              Py3.6+
+     2019              Py3.5+
+     2018              Py2.7, Py3.4+
```

Comment:
Wishlist, optional: Would be nice to have scipy versions in addition to the years.

Reply:
That info is already present in the next table right below this one.